app/src/main/java/com/firebase/uidemo/database/firestore/FirestorePagingActivity.java

```diff
@@ -0,0 +1,194 @@
+package com.firebase.uidemo.database.firestore;
+
+import android.arch.paging.PagedList;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.firebase.ui.firestore.paging.FirestorePagingAdapter;
+import com.firebase.ui.firestore.paging.FirestorePagingOptions;
+import com.firebase.ui.firestore.paging.LoadingState;
+import com.firebase.uidemo.R;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.Query;
+import com.google.firebase.firestore.WriteBatch;
+
+import java.util.Locale;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class FirestorePagingActivity extends AppCompatActivity {
+
+    private static final String TAG = "FirestorePagingActivity";
+
+    @BindView(R.id.paging_recycler)
+    RecyclerView mRecycler;
+
+    @BindView(R.id.paging_loading)
+    ProgressBar mProgressBar;
+
+    private FirebaseFirestore mFirestore;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_firestore_paging);
+        ButterKnife.bind(this);
+
+        mFirestore = FirebaseFirestore.getInstance();
```

Comment:
I'm wondering what the best practices should be... I think we should store the collection as a field, but use the full `FirebaseFirestore.getInstance()` for other stuff like the batch. And make it final/inline the assignment? Yay, nay?

Reply:
Gonna do all of that besides the inline assignment (not a fan of that in classes where you don't have a constructor, like `Activity` or `Fragment`)